internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -258,3 +267,23 @@ func (r *Reconciler) getNode(ctx context.Context, c client.Reader, providerID *n
 
 	return &nodeList.Items[0], nil
 }
+
+func (r *Reconciler) reconcileNodeTaints(ctx context.Context, remoteClient client.Client, node *corev1.Node) error {
+	taints := []corev1.Taint{}
+	patchHelper, err := patch.NewHelper(node, remoteClient)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create patch helper for Node %s", klog.KObj(node))
+	}
+	// Drop the NodeUninitializedTaint taint on the node.
+	for _, taint := range node.Spec.Taints {
+		if taint.MatchTaint(&clusterv1.NodeUninitializedTaint) {
+			continue
+		}
+		taints = append(taints, taint)
+	}
+	node.Spec.Taints = taints
+	if err := patchHelper.Patch(ctx, node); err != nil {
```

Comment:
Also in this case, wondering if we can do
```
if taints.RemoveTaints(node, desired) {
    patch...
}
```
And issue the patch only if there are changes

Reply:
Added a new helper for this.